examples/factors/README.md

```diff
@@ -57,14 +55,14 @@ Don't worry about why these lines are included yet; for now, we're just being di
 Here are what the other two lines with `METHOD_NAME_ELF` and `METHOD_NAME_ID` should look like after updating:
 
 ```rust
-    let mut prover =
-        Prover::new(MULTIPLY_ELF).expect("Prover should be constructed from valid ELF binary");
-
+fn main() {
+    // Here we make an executor, loading the (renamed) ELF binary.
+    let mut exec = Executor::from_elf(env, METHOD_NAME_ELF).unwrap();
```

Comment:
This takes an `env` parameter, which isn't defined anywhere.

Here's a suggestion, let's figure out how to make this README something that gets run as part of doc builds, which would then give us the ability to 'unit test' this example code.

Reply:
Looking into this and working out errors right now -- before I continue, is this an improvement we should be aiming to include in today's release?

Reply:
I think we should try to get as much of our docs under test as possible

Reply:
Looking at this thread: https://github.com/rust-lang/cargo/issues/383, it seems like built in Rust tooling won't accomplish this. It includes a comment suggesting a fairly straightforward macro for adding README tests: https://github.com/rust-lang/cargo/issues/383#issuecomment-720873790, however I haven't gotten this working -- I'll take a less rushed look tomorrow.

Reply:
That's the one I've been trying out and running into errors, going to keep troubleshooting today.

Reply:
More detailed version: `#![doc = include_str!("../README.md")]` is included in most of our crates under `src/lib.rs`, but I'm getting fingerprint errors that seem to be complaining about files changing out from under the compiler when I add the line to the factors example under `src/main.rs`. I'm going to pull my branch down and rerun in case I've done something funny locally, but it's not yet complaining about the syntax of various README snippets, my marker for at least having the README successfully included.

Reply:
The docs are building now so the branch issue was probably just git flow cruft but it's strange that only the digital signature example README is sensitive to asserting untruths.

Reply:
From convo with @flaub for now will add factors function and tests into a `lib.rs` to "solve" this problem but will come back to this (and see what else is out there with untested README snippets).

Reply:
Working through decisions around how to present code snippets to get this functional but still educational. The one that's most displeasing is aligning with our per-example naming conventions. We change `methods` to `factors-methods`, and thus
`use methods::{MULTIPLY_ID, MULTIPLY_ELF};` fails as an import unless I tell people to first complete this rename and get `use factor_methods::{MULTIPLY_ID, MULTIPLY_ELF};`.

I love that we've cut a clean path in our README to "minimal steps necessary to successfully build" to get people into the edit-compile loop as quickly as possible. Injecting this rename task breaks that a tiny bit, but I'm writing out instructions to feel it out.

Reply:
A second sub-ideal situation is that the final product is now split into `main.rs` and `lib.rs`, so there's not a pleasant 1:1 to follow along with. That said, the cost of having outdated README here is sufficiently high to warrant proceeding. There might be room to tack on a "let's make a library" step. It just won't give folks an instant visual clue. 

Loving @tzerrell 's suggestion to just include the README and nothing else. Tradeoff here is not teaching "let's make a library" so I'll leave it off for now but it's an option to consider if we strongly dislike the complexity of this next revision.